src/instrumenter/instrument.ts

```diff
@@ -154,22 +153,32 @@ function getDebugInfoEmits(
             );
         }
 
-        const assertionFailedDataExpr = gt(instrCtx.compilerVersion, "0.6.2")
-            ? instrCtx.getAssertionFailedDataEvent(annot.target)
-            : instrCtx.getAssertionFailedDataFun(annot.target);
-
         // Finally construct the emit statement for the debug event.
-        const emitStmt = makeEmitStmt(instrCtx, assertionFailedDataExpr, [
-            factory.makeLiteral("int", LiteralKind.Number, "", String(annot.id)),
-            factory.makeFunctionCall(
-                "<missing>",
-                FunctionCallKind.FunctionCall,
-                factory.makeIdentifier("<missing>", "abi.encode", -1),
-                evtArgs
-            )
-        ]);
+        if (instrCtx.assertionMode === "hardhat") {
+            const emitStmts: Statement[] = [];
+
+            for (const evtArg of evtArgs) {
```

Comment:
Why are we logging the arguments one by one? Can't we emit a single log statement with all the event args (and maybe the event location/name?

Reply:
In #114 there is an entry:

> Note that the `console.log()` functions in hardhat are limited to up to 4 arguments, so extra work needs to be done when we need to print more than 4 values.

It seems cheaper to implement standalone logging. Otherwise we whould have to chunk-split the event args and emit `console.log()` per each pair. I'm also not sure if event args are always instance of `Identifier` to also write their names. WDYT?

Reply:
Partially fixed in 4e6e130

Reply:
Again, removed "chunking" logic due to poor support of `bytes memory` in overloaded `console.log()`. It would be a significant logic compilcation to decide if Scribble should use over `console.log()` that supports very limited se of primitive args, or specific `console.log*()` function for standalone argument. Give that there is a possibility to hit compliex types and wrap them around with `abi.encode()`, we better stick to single solution. See bf8a65a.